### PR TITLE
Feature/testing model

### PR DESCRIPTION
**Pull request by rmudambi**

### CODE REVIEW

File: src/vivarium_csu_swissre_cancer/utilities.py [hunk 1]
```diff
@@ -1,23 +1,53 @@
...
     """
-    return len(max(project_globals.LOCATIONS, key=len))
+    def __init__(self, name, mean, sd, lower=0, upper=1):
+        self.name = name
+        self.a = (lower - mean) / sd if sd else 0
+        self.b = (upper - mean) / sd if sd else 0
+        self.loc = mean
+        self.scale = sd
+        
+    def sample_screening_parameter(self, draw: int) -> float:
```
Comment (collijk) [inline src/vivarium_csu_swissre_cancer/utilities.py:30]: Probably shouldn't be specific to screening

File: src/vivarium_csu_swissre_cancer/components/screening.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+        builder.event.register_listener('time_step',
+                                        self.on_time_step)
+
+    def on_initialize_simulants(self, pop_data: 'SimulantData'):
+        """Assign all simulants a next screening date. Also determine if they attended their previous screening"""
+
+        pop = self.population_view.subview([
+            SEX,
+            AGE,
+            project_globals.BREAST_CANCER_MODEL_NAME,
```
Comment (collijk) [inline src/vivarium_csu_swissre_cancer/components/screening.py:94]: check this out for framework error.

In practice, get a subview containing only the columns you need

File: src/vivarium_csu_swissre_cancer/globals.py [hunk 3]
```diff
@@ -137,18 +138,39 @@ def __new__(cls, value):
...
+    MRI_SENSITIVITY: TruncnormDist = TruncnormDist('mri_sensitivity', 0.91, 0.0091)
+    MRI_SPECIFICITY: TruncnormDist = TruncnormDist('mri_specificity', 1.0, 0.0)
+    
+    ULTRASOUND_SENSITIVITY: TruncnormDist = TruncnormDist('ultrasound_sensitivity', 0.737, 0.00737)
+    ULTRASOUND_SPECIFICITY: TruncnormDist = TruncnormDist('ultrasound_specificity', 1.0, 0.0)
+
+    MAMMOGRAM_ULTRASOUND_SENSITIVITY: TruncnormDist = TruncnormDist('mammogram_ultrasound_sensitivity', 0.939, 0.00939)
+    MAMMOGRAM_ULTRASOUND_SPECIFICITY: TruncnormDist = TruncnormDist('mammogram_ultrasound_specificity', 1.0, 0.0)
+
+    BASE_PROBABILITY: TruncnormDist = TruncnormDist('probability_attending_screening', 0.3, 0.003)
```
Comment (rmudambi) [inline src/vivarium_csu_swissre_cancer/globals.py:160]: These probabilities should have the same key

File: src/vivarium_csu_swissre_cancer/components/screening.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+        """
+        self.clock = builder.time.clock()
+        self.step_size = builder.time.step_size()
+        self.randomness = builder.randomness.get_stream(self.name)
+
+        draw = builder.configuration.input_data.input_draw_number
+        self.screening_parameters = {parameter.name: parameter.get_random_variable(draw)
+                                     for parameter in project_globals.SCREENING}
+
+        required_columns = [AGE, SEX, project_globals.BREAST_CANCER_MODEL_NAME]
```
Comment (collijk) [inline src/vivarium_csu_swissre_cancer/components/screening.py:64]: Don't need breast cancer

File: src/vivarium_csu_swissre_cancer/components/screening.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+        sensitivity[ultrasound] = self.screening_parameters[project_globals.SCREENING.ULTRASOUND_SENSITIVITY.name]
+        sensitivity[mammogram_ultrasound] = self.screening_parameters[
+            project_globals.SCREENING.MAMMOGRAM_ULTRASOUND_SENSITIVITY.name
+        ]
+        sensitivity[mammogram] = self.screening_parameters[project_globals.SCREENING.MAMMOGRAM_SENSITIVITY.name]
+
+        # TODO add in specificity if the specificity ever changes from 100%
+
+        # Perform screening on those who attended screening
+        accurate_results = self.randomness.get_draw(pop.index) < sensitivity
```
Comment (collijk) [inline src/vivarium_csu_swissre_cancer/components/screening.py:184]: need additional key

File: src/vivarium_csu_swissre_cancer/components/screening.py [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+        # Get all simulants with a screening scheduled during this timestep
+        pop = self.population_view.get(event.index)
+        screening_scheduled_mask = pop[project_globals.NEXT_SCREENING_DATE] < self.clock()
+        screening_scheduled = pop[screening_scheduled_mask]
+
+        # Get probability of attending the next screening for scheduled simulants
+        p_attends_screening = self._get_screening_attendance_probability(screening_scheduled)
+
+        # Get all simulants who actually attended their screening
+        attended_this_screening = self.randomness.get_draw(screening_scheduled.index) < p_attends_screening
```
Comment (collijk) [inline src/vivarium_csu_swissre_cancer/components/screening.py:126]: This is the other randomness thing without additional key

**Pull request merged**